coordinator/provider/keyranges.go

```diff
@@ -62,13 +63,36 @@ func (c CoordinatorService) KeyRangeIDByBounds(ctx context.Context, keyRange *pr
 }
 
 func (c CoordinatorService) SplitKeyRange(ctx context.Context, request *protos.SplitKeyRangeRequest) (*protos.ModifyReply, error) {
-	err := c.impl.Split(ctx, &kr.SplitKeyRange{
-		Bound: request.Bound,
-	})
+	krsqb, err := c.impl.ListKeyRange(ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	splitKR := &kr.SplitKeyRange{
+		Bound: request.Bound,
+		Krid:  request.KeyRangeInfo.Krid,
+	}
+
+	// TODO: choose a key range without hacks.
```

Comment:
lets just explisitly pass source key range here (and regenerate protos accordingly)

Reply:
Fixed